core/common/src/main/java/alluxio/exception/AlluxioException.java

```diff
@@ -77,23 +77,30 @@ public AlluxioTException toThrift() {
    * @return the native Alluxio exception
    */
   public static AlluxioException fromThrift(AlluxioTException e) {
-    try {
-      Class<? extends AlluxioException> throwClass;
-      if (e.isSetClassName()) {
-        // server version 1.1.0 or newer
+    Class<? extends AlluxioException> throwClass;
+    if (e.isSetClassName()) {
+      // server version 1.1.0 or newer
+      try {
         throwClass = (Class<? extends AlluxioException>) Class.forName(e.getClassName());
-      } else {
-        // server version 1.0.x
-        throwClass = AlluxioExceptionType.getAlluxioExceptionClass(e.getType());
-      }
-      if (throwClass == null) {
-        throwClass = AlluxioException.class;
+      } catch (ClassNotFoundException ee) {
+        // this can happen when the client is talking to a newer version of a server that
+        // introduced an exception that the client does not recognize
+        throwClass = UnexpectedAlluxioException.class;
       }
+    } else {
+      // server version 1.0.x
+      throwClass = AlluxioExceptionType.getAlluxioExceptionClass(e.getType());
+    }
+    if (throwClass == null) {
```

Comment:
Should this be in the `else` block? Or is it possible for the first `if` block to set `throwClass` to null?